R/prepare-lpr.R

```diff
@@ -93,6 +104,19 @@ prepare_lpr3 <- function(kontakter, diagnoser) {
   ) |>
     logic_as_expression()
 
+  # Consider LPR_A data only if it is specified in the input:
+  if (!is.null(lpr_a_kontakt) && !is.null(lpr_a_diagnose)) {
```

Comment:
For example, by using the same function for LPR3 and LPRA, you've made the implementation complicated by including a control flow related to the optional arguments. Whenever you add control flows like if else, you increase complexity of code. The less complexity, the easier it is to maintain and read.